### ISSUE TITLE
enhanceTextSelection not working

### ISSUE THREAD

**Issue by Vi-dot** [mozilla/pdf.js#9943]
Attach (recommended) or Link to PDF file here:
Any pdf, but here is an example : 
[Rigby_2018_AJ_155_104.pdf](https://github.com/mozilla/pdf.js/files/2244521/Rigby_2018_AJ_155_104.pdf)

Configuration:
- Web browser and its version: Any firefox or chrome but for now I test on Firefox 61.0.1 and Chrome 67.0.3396.99
- Operating system and its version: Ubuntu 16.04 LTS
- PDF.js version: 2.0.550
- Is a browser extension: No

Steps to reproduce the problem:
```typescript
  public setupViewer() {

    PDFJS.disableTextLayer = !this._renderText;

    let firstDiv = this.element.nativeElement.querySelector('div');

    let pdfOptions = {
      container: this._container || firstDiv,
      viewer: firstDiv.firstElementChild
    };

    if (this._renderLink) {
      this._pdfLinkService = new PDFJSViewer.PDFLinkService();
      pdfOptions['linkService'] = this._pdfLinkService;
    }

    if (!this._pageBorder) {
      pdfOptions['removePageBorders'] = true;
    }

    if (this._enhanceTextSelection) {
      pdfOptions['enhanceTextSelection'] = true;
      //pdfOptions['textLayerMode'] = 2;
    }
    else {
      pdfOptions['enhanceTextSelection'] = false;
      //pdfOptions['textLayerMode'] = 0;
    }

    this.zone.runOutsideAngular(() => {
      this._pdfViewer = new PDFJSViewer.PDFViewer(pdfOptions);
    });

    if (this._renderLink) {
      this._pdfLinkService.setViewer(this._pdfViewer);
    }

    if (this._finder) {
      this._pdfFindController = new PDFJSViewer.PDFFindController({
        pdfViewer: this._pdfViewer
      });
      this._pdfViewer.setFindController(this._pdfFindController);

      this._pdfFindController.updateUIResultsCount = () => {
        this.emitSearchTextDone();
      };
    }

    if (this._src) {
      this.loadPDF();
    }
  }

  public loadPDF(src?) {

    if (!src) {
      src = this._src;
    }

    if (src) {

      this.zone.runOutsideAngular(() => {
      PDFJS.getDocument(src).then((pdf: PDFDocumentProxy) => {

        this._pdf = pdf;
        this.lastLoaded = src;

        if (this.afterLoadComplete && typeof this.afterLoadComplete === 'function') {
          this.afterLoadComplete(pdf);
        }

        this._pdfViewer.setDocument(this._pdf);
        this._pdfLinkService.setDocument(this._pdf, null);

        this.render();
      });
      });
    }
  }

  public render() {
    if (!this.isValidPageNumber(this._page)) {
      this._page = 1;
    }

    if (this._rotation !== 0 || this._pdfViewer.pagesRotation !== this._rotation) {
      this._pdfViewer.pagesRotation = this._rotation;
    }

    if (this._stickToPage) {
      this._pdfViewer.currentPageNumber = this._page;
    }

    this.updateSize();
  }

  public updateSize() {
    if (!this._originalSize) {
      if (this._pdf) {
        this.zone.runOutsideAngular(() => {
        this._pdf.getPage(this._pdfViewer._currentPageNumber).then((page: PDFPageProxy) => {
          this.curScale = this._zoom * (this.element.nativeElement.offsetWidth / page.getViewport(1).width) / PdfViewerComponent.CSS_UNITS;
          this._pdfViewer._setScale(this.curScale, !this._stickToPage);
        });
        });
      }
    } else {
      this.curScale = this._zoom;
      this._pdfViewer._setScale(this.curScale, !this._stickToPage);
    }
  }
```

What is the expected behavior?
enhanceTextSelection option should improve text selection.

What went wrong? (add screenshot)
If I enable/disable enhanceTextSelection, nothing changes, text selection is bad when white spaces are big or when we start selection outside of the paragraph. 
We use it for annotating, then it would be a big improvement for us.

Thanks

**Comment by timvandermeij** [comment]
The `enhanceTextSelection` option is deprecated; you should use `textLayerMode = 2` (refer to https://github.com/mozilla/pdf.js/blob/master/web/ui_utils.js#L41). Other than that, I can't directly see why this would not work, but please note that in the issue template we describe that just code snippets won't help to look into the issue. Please provide a running version on e.g., JSBin, JSFiddle or your own server.

**Comment by Snuffleupagus** [comment]
In addition to everything already mentioned in https://github.com/mozilla/pdf.js/issues/9943#issuecomment-409369485, the `PDFJS` global was removed in version `2.0` of the library. Provided that the configuration information, as given above, is accurate:

> PDF.js version: 2.0.550

attempting to run the snippet above ought to throw a `ReferenceError`.

**Comment by Vi-dot** [comment]
Hello,
Sorry for the response time.
Here is the [JSFiddle](https://jsfiddle.net/Vi_dot/25nq4kt8/4/)
(With last version of pdf.js : 1.10.100)

I tested with **Chrome** and it works great.
But with **Firefox 61.0.1**, `enhanceTextSelection` or `textLayerMode=2` changes nothing.

**Comment by timvandermeij** [comment]
Your fiddle works just fine for me in Firefox 61.0.2. Since you're using version 1.10 and not version 2.0, you may only need `enhanceTextSelection` since I think `textLayerMode` is not yet introduced in that version. If I removed it in your fiddle, it also works. Note that the enhanced text selection means that you can for example start dragging next to the title (after `Dynamic`) and it will start the selection. If you do that at https://mozilla.github.io/pdf.js/web/viewer.html, it won't select anything.

In short, if this doesn't work for you, could you point out what exactly is not working, as your fiddle works fine for me?

**Comment by Vi-dot** [comment]
Thanks,
If I understood well, with enhanced text selection, we can start selection outside of a paragraph and it will continue on the nearest text. It sounds very usefull for our use case : annotating pdf.
We tested on few computers with Ubuntu (v 14/16/18), and last stable version of Firefox (v 61.0.1), it doesn't work.
But it works on Chrome, and I saw the difference when I enable or disable `enhanceTextSelection`.
The issue is that Firefox is our main web browser.

**Comment by Vi-dot** [comment]
I installed **Firefox 61.0.2**, and the issue is the same :/

**Comment by timvandermeij** [comment]
Yes, that is correct. I'm not really sure why your fiddle does work for me in Firefox and not for you though.

**Comment by Vi-dot** [comment]
I just tried the fiddle on an other computer with Windows 10 this time,
It still doesn't work with Firefox 61.0.2
It works perfectly with Chrome 68.0.3440.106 or with Microsoft Edge 42.17134.1.0

**Comment by timvandermeij** [comment]
Do you perhaps have content security policies enabled for Firefox? It could be that something is blocking this functionality. You could try opening Firefox with a clean profile to check if that is perhaps causing issues. I can't really think of anything else as to why this would not work in Firefox specifically.

**Comment by timvandermeij** [comment]
Closing as answered since the fiddle works.
